### ISSUE TITLE
Add option to clear bundle gem cache in bundle clean

### ISSUE THREAD

**Issue by wildmaples** [ruby/rubygems#7163]
### Feature request

The `bundle clean` command currently removes all the unused gems in the bundler directory. 

However, it would be nice to have an option for `bundle clean` to also remove the `.gem` files, perhaps something like `bundle clean --cache`. It would make it easier for us to clean up a docker container after bundle installing. For example, the command could replace this line in the [Rails dockerfile](https://github.com/rails/rails/blob/main/railties/lib/rails/generators/rails/app/templates/Dockerfile.tt#L45) that removes the cache folder. 

`bundle clean` used to have this behaviour in v1.1.6 (2011) but no longer in 2016 (issue: https://github.com/rubygems/bundler/issues/1293). 

cc: @colby-swandale

**Comment by rubyFeedback** [comment]
Not disagreeing about the feature itself, but I wonder if the --cache flag is descriptive of the described feature (removing .gem files). Would you consider another name as appropriate? Not sure I can come up with a short one; I tend to use soemwhat longer flags for my own gems, e. g. --remove-gem-file or something like that. Or, perhaps an alternative, something like "bundle clean --extensive" or something like that, and describe that this will also eliminate .gem files.

**Comment by wildmaples** [comment]
Thanks for the feedback, @rubyFeedback! Totally not committed to the flag's name - just a suggestion/example to start us off.

**Comment by deivid-rodriguez** [comment]
> bundle clean used to have this behaviour in v1.1.6 (2011) but no longer in 2016 (issue: https://github.com/rubygems/bundler/issues/1293).

Just to clarify, my understanding is that this behavior is still in Bundler, but that's only for gems outside of the current bundle. In other words, `bundle clean` already cleans up `.gem` files in cache for gems no longer present in the bundle.

If I'm understanding correctly, what's being requested here is a special flag to `bundle clean` to completely wipe out the cache.

**Comment by eregon** [comment]
Coming from https://github.com/ruby/setup-ruby/issues/645 it would be good to have a way for Bundler to remove `"${BUNDLE_PATH}"/$RUBY_ENGINE/*/cache/*.gem` files on `bundle install` (via some flag) since those files will not be necessary after and they can occupy a significant amount of disk space, as well as in the gem cache.

**Comment by deivid-rodriguez** [comment]
This is super related to #3225. I agree it'd be nice to have a builtin way to clean up the `.gem` packages, either through a `bundle install` or a `bundle clean` flag. Of course this can be workaround with a manual `rm -rf` but having a way to do it inside Bundler abstracts away the internal directory structure, so makes things easier for people.

**Comment by hsbt** [comment]
The `prune` setting covers this. `BUNDLE_PRUNE=cache` removes the download cache under the bundle path after a successful `bundle install`, `bundle update` or `bundle cache`, and on `bundle clean`. It only acts when a bundle path is set, so it never touches the cache RubyGems shares in the system gem directory. It ships in RubyGems 4.1.0. Please try it out once that release lands.
